src/foam/dao/EasyDAO.js

```diff
@@ -836,9 +853,9 @@ model from which to test ServiceProvider ID (spid)`,
       javaCode: `
         if ( getJournalType().equals(JournalType.SINGLE_JOURNAL) ) {
           if ( getWriteOnly() ) {
-            delegate = new foam.dao.WriteOnlyJDAO(x, delegate, getOf(), getJournalName());
+            delegate = new foam.dao.WriteOnlyJDAO(subx, delegate, getOf(), getJournalName());
```

Comment:
subx is never defined

Reply:
as discussed, this was refinement-related weirdness. changed back to "x"